src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs

```diff
@@ -22,8 +22,10 @@ namespace Microsoft.AspNetCore.OpenApi;
 /// Provides a set of extension methods for modifying the opaque JSON Schema type
 /// that is provided by the underlying schema generator in System.Text.Json.
 /// </summary>
-internal static class JsonObjectSchemaExtensions
+internal static class JsonNodeSchemaExtensions
 {
+    private static readonly NullabilityInfoContext _nullabilityInfoContext = new();
```

Comment:
`NullabilityInfoContext` isn't thread safe and hasn't been designed to be used as a singleton. I would recommend allocating a fresh instance for every type graph you are looking to generate.